File: Medium/2025-09-29-1039-Minimum-Score-Triangulation-of-Polygon/solution.py
class Solution:
    def minScoreTriangulation(self, values: List[int]) -> int:
        n = len(values)
        dp = [[0] * n for _ in range(n)]
        
        for length in range(2, n):  # length of the triangle base
            for i in range(n - length):
                j = i + length
                dp[i][j] = float('inf')
                for k in range(i + 1, j):
                    score = values[i] * values[k] * values[j]
                    dp[i][j] = min(dp[i][j], dp[i][k] + dp[k][j] + score)
        
        return dp[0][n - 1]

File: Medium/2025-09-29-1039-Minimum-Score-Triangulation-of-Polygon/solution.js
var minScoreTriangulation = function(values) {
    const n = values.length;
    const dp = Array.from({ length: n }, () => Array(n).fill(0));

    for (let len = 2; len < n; len++) {
        for (let i = 0; i < n - len; i++) {
            const j = i + len;
            dp[i][j] = Infinity;
            for (let k = i + 1; k < j; k++) {
                const score = values[i] * values[k] * values[j];
                dp[i][j] = Math.min(dp[i][j], dp[i][k] + dp[k][j] + score);
            }
        }
    }

    return dp[0][n - 1];
};